src/game/engine/building.cpp

```diff
@@ -131,5 +136,153 @@ void *BuildingClass::Get_Image_Data() const
     if (m_CurrentState == BSTATE_0) {
         return Class_Of().Get_Buildup_Data();
     }
-    return ObjectClass::Get_Image_Data();
+    return TechnoClass::Get_Image_Data();
+}
+
+/**
+ *
+ *
+ */
+void BuildingClass::Active_Click_With(ActionType action, ObjectClass *object)
+{
+    if (action == ACTION_ATTACK && object != nullptr) {
```

Comment:
These Active_Click_With functions should probably be a switch (action) block rather than an if ladder.